src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1500,7 +1500,7 @@ public Wallet GetWalletByName(string walletName)
         /// <inheritdoc />
         public ICollection<uint256> GetFirstWalletBlockLocator()
         {
-            return this.Wallets.First().BlockLocator;
+            return this.Wallets.FirstOrDefault()?.BlockLocator ?? new[] { this.network.GenesisHash };
```

Comment:
If there's no wallet, "GetFirstWalletBlockLocator" shouldn't even be called. If it is called and there is no wallet, I think it's better to return null or an empty collection.

Reply:
Unfortunately returning 0 items will cause the calling code to fail:

```
                ICollection<uint256> locators = this.walletManager.GetFirstWalletBlockLocator();
                var blockLocator = new BlockLocator { Blocks = locators.ToList() };
                ChainedHeader fork = this.chain.FindFork(blockLocator);
                this.walletManager.RemoveBlocks(fork);
                this.walletManager.WalletTipHash = fork.HashBlock;
                this.walletTip = fork;
```

`FindFork` would return null.

Reply:
the problem is that if the wallet tip is the genesis, you're going to be syncing from the beginning when you create a wallet.

Reply:
Also, the block locator is located in the wallet file, not sure what you'd do with a block locator if you don't have a wallet.

Reply:
Perhaps the way to think about this is that the wallet tip keeps track of the blocks we no longer need to deal with in order to keep our current set of wallets up-to-date.

That would imply that the tip can be the current chain tip in the case where there are no wallets.